lib/galaxy/tool_util/deps/container_classes.py

```diff
@@ -62,13 +61,19 @@
 tag, tag_value = next(iter(meta.items()))
 rev, rev_value = next(iter(tag_value.items()))
 cmd = "${images_cmd}"
-proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, text=True)
 stdo, stde = proc.communicate()
 found = False
 for line in stdo.split("\n"):
-    tmp = re.split(r'\s+', line)
-    if tmp[0] == tag and tmp[1] == rev and tmp[2] == rev_value:
-        found = True
+    line = line.strip()
+    if line:
+        try:
+            img = json.loads(line)
+            if img.get("Repository") == tag and img.get("Tag") == rev and img.get("ID") == rev_value:
```

Comment:
Variable naming is confusing here (but it was already before). 

Maybe we rename `tag` to `repository` and `rev` to `tag`? 

above

```diff
- tag, tag_value = next(iter(meta.items()))
- rev, rev_value = next(iter(tag_value.items()))
+ repository, _ = next(iter(meta.items()))
+ tag, id = next(iter(tag_value.items()))
```

```suggestion
            if img.get("Repository") == repository and img.get("Tag") == tag and img.get("ID") == id:
```

But we can also leave it. Up to you.

Reply:
That makes sense. :+1: